Add information about enabling FIPS (#537)
openstack-k8s-operators/data-plane-adoption: docs_user/modules/con_fips-support.adoc

[id="fips-support_{context}"]

= FIPS support

> If you are adopting an {OpenStackShort} {rhos_prev_ver} FIPS environment to {OpenStackShort} {rhos_curr_ver}, your adopted cluster will remain a FIPS cluster. There is no option to change FIPS status during adoption.

Reviewer: ```suggestion
If you are adopting a {rhos_prev_long} {rhos_prev_ver} FIPS environment to {rhos_long} {rhos_curr_ver}, your adopted cluster remains a FIPS cluster. There is no option to change FIPS status during adoption.
```
The first part without the attributes translates to "If you are adopting a Red Hat OpenStack Platform 17.1 FIPS environment to Red Hat OpenStack Services on OpenShift (RHOSO) 18.0..." Is that what you intended?

> There is a major difference with how FIPS was configured in {OpenStackPreviousInstaller} and how it is enabled and enforced in operator deployments. In {OpenStackPreviousInstaller} FIPS is enabled as part of its configuration, whereas in operator deployments, there is no specific FIPS configuration at the operator level.

Reviewer: ```suggestion
There is a major difference with how FIPS was configured in {OpenStackPreviousInstaller} and how it is enabled and enforced in operator deployments. In {OpenStackPreviousInstaller}, FIPS is enabled as part of its configuration, whereas in operator deployments, there is no specific FIPS configuration.
```

> If your cluster is FIPS enabled, you must deploy a FIPS {OpenShiftShort} cluster to adopt your {OpenStackShort} {rhos_prev_ver} FIPS control plane.

Reviewer: ```suggestion
If your cluster is FIPS enabled, you must deploy a FIPS {OpenShift} cluster to adopt your {OpenStackShort} {rhos_prev_ver} FIPS control plane.
```

> Enabling FIPS in {OpenShiftShort} is out of scope for this guide. Please refer to the link:https://docs.openshift.com/container-platform/latest/installing/installing-fips.html[Support for FIPS cryptography] in _OpenShift Container Platform Documentation_.

Reviewer: ```suggestion
For more information about enabling FIPS in {OpenShiftShort}, see link:https://docs.openshift.com/container-platform/latest/installing/installing-fips.html[Support for FIPS cryptography] in _OpenShift Container Platform Documentation_.
```

[id="fips-support_{context}"]

= FIPS support

There is a major difference with how FIPS was configured in {OpenStackPreviousInstaller} and how it is enabled and enforced in operator deployments.

> In {OpenStackPreviousInstaller} FIPS is enabled as part of it's configuration.

Reviewer 2: it's -> its

In operator deployments there is no specific FIPS configuration at the operator level. If your deployment has FIPS compliance requirements, the whole {OpenShift} ({OpenShiftShort}) cluster must be installed with FIPS being enabled. This will have happened before proceeding with this guide and installing {OpenStackShort}.

> 

Reviewer 2: Is it the case that if the source OpenStack deployment is non-FIPS, the destination OpenShift should be configured as non-FIPS too? (If yes then this is probably worth a note?)

Author: Thanks for the comment, I'll update it with JP's feedback: "If you are adopting an OSP 17.1 FIPS environment to RHOSO 18.0, your adopted cluster will remain a FIPS cluster (there is no option to change FIPS status during adoption). You must deploy a FIPS OCP cluster to adopt your OSP 17.1 FIPS control plane"

> 
> 
> There is a major difference with how FIPS was configured in {OpenStackPreviousInstaller} and how it is enabled and enforced in operator deployments.
> 
> In {OpenStackPreviousInstaller} FIPS is enabled as part of it's configuration.
> 
> In operator deployments there is no specific FIPS configuration at the operator level. If your deployment has FIPS compliance requirements, the whole {OpenShift} ({OpenShiftShort}) cluster must be installed with FIPS being enabled. This will have happened before proceeding with this guide and installing {OpenStackShort}.
> 
> Please refer to the link:https://docs.openshift.com/container-platform/latest/installing/installing-fips.html[Support for FIPS cryptography] in _OpenShift Container Platform Documentation_.

Reviewer: ```suggestion

There is a difference between how Federal Information Processing Standard (FIPS) was configured in {OpenStackPreviousInstaller} and how it is enabled and enforced in operator deployments.

* In {OpenStackPreviousInstaller}, FIPS is enabled as part of its configuration.
* In operator deployments, there is no specific FIPS configuration at the operator level. If your deployment has FIPS compliance requirements, you must install the whole {OpenShift} ({OpenShiftShort}) cluster with FIPS enabled. The installation happens before you proceed with the adoption and install {rhos_prev_long}.

For more information about FIPS support, see link:https://docs.openshift.com/container-platform/latest/installing/installing-fips.html[Support for FIPS cryptography] in _OpenShift Container Platform Documentation_.
```

Reviewer: When you say, "FIPS is enabled as part of its configuration". do you mean that FIPS is enabled by default? 
I yes, I think it would be clearer to say "In director, FIPS is enabled by default."

Reviewer 3: No, FIPS was not enabled by default.  Rather, there was an additional template that enabled certain THT parameters that were fips specific.

Author: I adjusted the wording, it should be more clear now.
